src/main/java/org/cryptomator/cloudaccess/api/CloudItemMetadata.java

```diff
@@ -42,4 +44,21 @@ public CloudItemType getItemType() {
 	public Optional<Long> getSize() {
 		return size;
 	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) return true;
+		if (o == null || getClass() != o.getClass()) return false;
+		CloudItemMetadata that = (CloudItemMetadata) o;
+		return Objects.equal(this.name, that.name) &&
+				Objects.equal(this.path, that.path) &&
+				this.itemType == that.itemType &&
+				Objects.equal(this.lastModifiedDate, that.lastModifiedDate) &&
+				Objects.equal(this.size, that.size);
```

Comment:
I prefer leading operators in multi-line statements:

```suggestion
		return Objects.equal(this.name, that.name)
				&& Objects.equal(this.path, that.path)
				&& this.itemType == that.itemType
				&& Objects.equal(this.lastModifiedDate, that.lastModifiedDate)
				&& Objects.equal(this.size, that.size);
```

Just my personal preference, nothing that needs to be changed 😉.

Reply:
Oh too fast, I will change this later on the dev branch 😅